virtcontainers/ich.go

```diff
@@ -0,0 +1,813 @@
+		UdsPath: 	ich.socketPath,
+		Port: 		1024,
+	}
+	ichConfig.Devices, err = ich.arch.appendHybridVSock(ichConfig.Devices, vsockDev)
+	if err != nil {
+		return err
+	}
+	
+	// Add the root disk to the hypervisor
+	imagePath, err := ich.config.ImageAssetPath()
+	if err != nil {
+		return err
+	}
+
+	if imagePath != "" {
+		ichConfig.Devices, err = ich.arch.appendImage(ichConfig.Devices, imagePath)
+		if err != nil {
+			return err
+		}
+	}
+	 
+	// Add the virtio-fs to the hypervisor
+	vfsdSockPath, err := ich.vfsdFSSocketPath(ich.id)
+	if err != nil {
+		return err
+	}	
+	if vfsdSockPath != "" {
+		ichConfig.Devices, err = ich.arch.appendVirtualFilesystem(ichConfig.Devices, "kataShared", vfsdSockPath)
+		if err != nil {
+			return err
+		}
+	}
+	
+	// Add the serial console to the cloud hypervisor
+	if ich.config.Debug {
+		ichConfig.Devices, err = ich.arch.appendSerialConsole(ichConfig.Devices, "file=/tmp/ich.log")
+	} else {
+		ichConfig.Devices, err = ich.arch.appendSerialConsole(ichConfig.Devices, "off")
+	}
+	if err != nil {
+		return err
+	}
+	
+	// Add the virtio console to the cloud hypervisor
+	ichConfig.Devices, err = ich.arch.appendVirtioConsole(ichConfig.Devices, "off")
+	if err != nil {
+		return err
+	}
+	 
+	// Add the http api endpoint to the cloud hypervisor by default
+//	apiSocketPath, err := ich.apiSocketPath(ich.id)
+//	if err != nil {
+//		return err
+//	}
+	
+//	ichConfig.Devices, err = ich.arch.appendApiSocket(ichConfig.Devices, apiSocketPath)
+//	if err != nil {
+//		return err
+//	}
+
+		
+	ich.ichConfig = ichConfig
+	return nil
+}
+
+func (ich *cloudHypervisor) startSandbox(timeout int) error {
+	span, _ := ich.trace("startSandbox")
+	defer span.Finish()
+	
+	ich.Logger().WithField("function", "startSandbox").Info("starting Sandbox")
+
+	vmPath := filepath.Join(store.RunVMStoragePath(), ich.id)
+	err := os.MkdirAll(vmPath, store.DirMode)
+	if err != nil {
+		return err
+	}
+	ich.Logger().WithField("function", "startSandbox").Infof("Starting virtiofsd") 
+	if ich.config.SharedFS == config.VirtioFS {
+		timeout, err = ich.setupVirtiofsd(timeout)
+		if err != nil {
+			return err
+		}
+		if err = ich.storeState(); err != nil {
+			return err
+		}
+	}
+	
+	ich.Logger().WithField("function", "startSandbox").Infof("virtiofsd starts sharing") 
+	
+	var strErr string
+	var thePid = 0
+	strErr, err, thePid = govmmIch.LaunchIch(ich.ichConfig)
+	if err != nil {
+		return fmt.Errorf("fail to launch cloud-hypervisor: %s, error messages from log: %s", err, strErr)
+	}		
+	if err := ich.waitVMM(ichTimeout); err != nil {
+		ich.Logger().WithField("cloud-hypervisor init failed:", err).Warn()
+		return err
+	}
+
+	ich.info.PID = thePid
+	ich.state.PID = thePid
+	ich.state.set(ichReady)
+	ich.storeState()
+
+	return nil
+}
+
+func (ich *cloudHypervisor) getSandboxConsole(id string) (string, error) {
+	ich.Logger().WithField("function", "getSandboxConsole").WithField("ID", id).Info("Get Sandbox Console")
+	// TODO
```

Comment:
Please can you add in a GitHub issue URL here so that the comment is in the form:

```go
// FIXME: https://github.com/kata-containers/runtime/issues/...
```

That way we have both the issue and the standardised comment as a reminder of what remains to be done.